ff/smirnoff_converter.py

```diff
@@ -85,95 +86,390 @@ def parse_quantity(number_string):
-            kb = parse_quantity(s.attributes['k'].value)
-            params.append([patt, kb, b0])
-        bonds = {
-            "params": params,
-        }
-        forcefield[tag] = bonds
-
-    elif tag == ANGLE_TAG:
-        params = []
-        for s in itemlist:
-            patt = s.attributes['smirks'].value
-            a0 = parse_quantity(s.attributes['angle'].value)
-            ka = parse_quantity(s.attributes['k'].value)
-            params.append([patt, ka, a0])
-        angles = {
-            "params": params,
-        }
-        forcefield[tag] = angles
-    elif tag == PROPER_TAG:
-        params = []
-        for s in itemlist:
-            patt = s.attributes['smirks'].value
-            counter = 1
-            components = []
-            while True:
-                try:
-                    k = parse_quantity(s.attributes['k'+str(counter)].value)
-                    phase = parse_quantity(s.attributes['phase'+str(counter)].value)
-                    period = float(s.attributes['periodicity'+str(counter)].value)
-                    idivf = float(s.attributes['idivf'+str(counter)].value)
-                    k = k/idivf
-                    components.append([k, phase, period])
-                    counter += 1
-                except KeyError:
-                    break
-            params.append([patt, components])
-        torsions = {
-            "params": params,
-        }
-        forcefield[tag] = torsions
-    elif tag == IMPROPER_TAG:
-        params = []
-        for s in itemlist:
-            patt = s.attributes['smirks'].value
-            impdivf = 3
-            k = parse_quantity(s.attributes['k1'].value)/impdivf
-            phase = parse_quantity(s.attributes['phase1'].value)
-            period = float(s.attributes['periodicity1'].value)
-            params.append([patt, k, phase, period])
-        impropers = {
-            "params": params
-        }
-        forcefield[tag] = impropers
-    elif tag == VDW_TAG:
-        params = []
-        for s in itemlist:
-            patt = s.attributes['smirks'].value
-            epsilon = parse_quantity(s.attributes['epsilon'].value)
-            rmin_half = parse_quantity(s.attributes['rmin_half'].value)
-            sigma = 2. * rmin_half / (2.**(1. / 6.))
-            params.append([patt, sigma, epsilon])
-        vdws = {
-            "params": params,
-            "props": {}
-        }
-        for key, val in xmldoc.getElementsByTagName("vdW")[0].attributes.items():
-            if key == 'cutoff':
-                # we don't do cuttoffs.
-                continue
-            elif 'scale' in key:
-                val = float(val)
-            elif key == 'switch_width':
-                continue
-            if key == "version":
-                continue
-            vdws["props"][key] = val 
-        forcefield['vdW'] = vdws
+AM1CCC_CHARGES = {'patterns': [('[#6X4:1]-[#16X1-1:2]', 0.9818644744993272),
```

Comment:
Perhaps it is overkill to add this to the script, it just seemed cleaner than having to copy it over by hand

Reply:
Would it be preferable if this dictionary were stored in a separate named file that is loaded by `ff/smirnoff_converter.py`, rather than being stored inside `ff/smirnoff_converter.py`?

(I don't know if it's important to store information about `AM1CCC_CHARGES` origin or versions, or if these are constrained to be identical to the AM1CCC definitions in `ff/params`)

Reply:
@proteneer What do you think? I am for moving this to another file, though I don't have a good answer regarding if we need to store these charges and if there are versions?

Reply:
Addressed in https://github.com/proteneer/timemachine/pull/491/commits/d755da86beb9fe268dc9f01c23ade01fbb18cd17